js/src/admin/components/MailPage.js

```diff
@@ -108,6 +105,28 @@ export default class MailPage extends Page {
     );
   }
 
+    getMailSettingsElementForm(field) {
+
+    if (field === 'mail_mailgun_endpoint') {
+
+      let endpoints = {
+        'api.eu.mailgun.net': 'EU Endpoint',
+        'api.mailgun.net': 'US Endpoint',
+      };
```

Comment:
I would prefer not to have this field-specific code here in the frontend, as the rest is so nicely generic.

Instead, we will need the API to differentiate between text fields and select boxes. (That means, the drivers will need this option in their `availableSettings()` method as well.)